### ISSUE TITLE
`W0612` issued for global variable even if their name matches `dummy-variables-rgx`, is this intended?

### ISSUE THREAD

**Issue by DervishD** [PyCQA/pylint#10890]
### Question

In this very minimal code:
````
# This has to appear OUTSIDE functions, `_` is a global variable.
_  = []
````

`W0612` is issued, marking `_`  as an unused varible, even though its name matches `dummy-variables-rgx`. In fact, the exact same variable is not marked as unused inside a function.

I think this is an intended behaviour related to `allow-unused-global-variables`, maybe a side effect of not having that option enabled, where ALL unused global variables are marked even if they match `dummy-variables-rgx`. I can't find this in the docs, and if it is intended, I think it should be documented.

### Documentation for future user

The information should be on the `variables` checker, showing that `allow-unused-global-variables` being `false` has more precendence than the variable name matching the `dummy-variables-rgx` regular expression.

**Comment by DanielNoord** [comment]
https://github.com/pylint-dev/pylint/blob/71caace29cd0fd42881f8b445728821dab665333/pylint/checkers/variables.py#L3272-L3289

I think we simply don't check the dummy variables rgx for these (yet)

**Comment by DervishD** [comment]
Yes, that's why I did not report that as a bug, but a documentation mismatch. If this is intended, rather than not cheking the rgx by mistake, I think it should be documented to avoid surprises. At first I thought that something was wrong with the example code I was checking, then I thought it was a configuration problem, etc.

Thanks for the clarification!

**Comment by DanielNoord** [comment]
Think this is indeed a bug.

**Comment by DervishD** [comment]
You know the intention better than me, so if you think it is a bug, is OK to me 👍🏼  Thanks for the help! 😄

**Comment by DervishD** [comment]
Wow, that was **FAST**!! Thanks a lot!
